### PR TITLE
(feat)add application level cpu chaos experiment

### LINKED ISSUE #974: Create resource exhaustion (CPU) chaos experiment at container level
Today, the resource exhaustion chaos experiment (generic/cpu-hog) has been implemented at a node-level. This needs to be implemented at a container level as well. This is a significant requirement, where the idea is to obtain the impact of resource chaos on targeted applications without increasing the blast radius of the experiment itself. 

Some of the possible approaches include: 

- Injecting a resource-exhaustion process into the target container which may run stress/stress-ng tools
- Inhibiting resources to the target container by limiting/patching minimal resources against it

### PR DESCRIPTION
**Pull request by ksatchit**
Signed-off-by: ksatchit <karthik.s@mayadata.io>

**What this PR does / why we need it**:

- Introduces a container-level/app-level CPU chaos experiment that can take desired number of cores as input against a target container & effect nearly ~90%  CPU consumption. 
- The experiment uses a custom kubernetes job that runs a docker-in-docker container to trigger the CPU chaos on the target container for a specified duration, returning the resources once complete.
- For more info on the chaos tool used, refer:  https://github.com/litmuschaos/test-tools/pull/108

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #974 

**Checklist**

* [ ] Does this PR have a corresponding GitHub issue?
* [ ] Have you included relevant README for the chaoslib/experiment with details?
* [ ] Have you added debug messages where necessary? 
* [ ] Have you added task comments where necessary? 
* [ ] Have you tested the changes for possible failure conditions?
* [ ] Have you provided the positive & negative test logs for the litmusbook execution?
* [ ] Does the litmusbook ensure idempotency of cluster state?, i.e., is cluster restored to original state?
* [ ] Have you used non-shell/command modules for Kubernetes tasks?
* [ ] Have you (jinja) templatized custom scripts that is run by the litmusbook, if any? 
* [ ] Have you (jinja) templatized Kubernetes deployment manifests used by the litmusbook, if any?
* [ ] Have you reused/created util functions instead of repeating tasks in the litmusbook?
* [ ] Do the artifacts follow the appropriate directory structure? 
* [ ] Have you isolated storage (eg: OpenEBS) specific implementations, checks? 
* [ ] Have you isolated platform (eg: baremetal kubeadm/openshift/aws/gcloud) specific implementations, checks?  
* [ ] Are the ansible facts well defined? Is the scope explicitly set for playbook & included utils?
* [ ] Have you ensured minimum/careful usage of shell utilities (awk, grep, sed, cut, xargs etc.,)?
* [ ] Can the limtusbook be executed both from within & outside a container (configurable paths, no hardcode)?
* [ ] Can you suggest the minimal resource requirements for the litmusbook execution?
* [ ] Does the litmusbook job artifact carry comments/default options/range for the ENV tunables?
* [ ] Has the litmusbooks been linted? 

**Special notes for your reviewer**:

### CODE REVIEW

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+          app_pod: "{{ app_pod_name.stdout }}"
+
+      when: "app_pod is not defined"
+
+    # here app_ns is the namespace of pod on which we are performing cpu chaos 
+    # in genric experiments app_ns is same as a_ns
+    # in openebs experiments app_ns is the namespace where openebs is installed i.e, openebs
+    - name: Identify the application node
+      shell: >
+        kubectl get pod {{ app_pod }} -n {{ app_ns }}
```
Comment (chandankumar4) [inline chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: Can you explain why we are using `a_ns` (line no 8) and `app_ns` (line no - 26)

File: experiments/generic/pod_cpu_hog/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+<table>
+<tr>
+<th> Name </th>
+<th> Description </th>
+<th> Documentation Link </th>
+</tr>
+<tr>
+ <td> Pod CPU Hog </td>
+ <td> This experiment causes CPU resource consumption on specified application containers by starting one or more md5sum calculation process on the special file /dev/zero. It Can test the application's resilience to potential slowness/unavailability of some replicas due to high CPU load.</td>
+ <td>  <a href="https://docs.litmuschaos.io/docs/pod-cpu-hog/"> Here </a> </td>
```
Comment (chandankumar4) [inline experiments/generic/pod_cpu_hog/README.md]: Instead of word `Here` can we use `Click here`

**Comment by ksatchit** [comment]
```
ansible-playbook 2.7.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 2.7.17 (default, Nov  7 2019, 10:07:09) [GCC 7.4.0]
Using /etc/ansible/ansible.cfg as config file
/etc/ansible/hosts did not meet host_list requirements, check plugin documentation if this is unexpected
/etc/ansible/hosts did not meet script requirements, check plugin documentation if this is unexpected
statically imported: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_prerequisites.yml
PLAYBOOK: pod_cpu_hog_ansible_logic.yml 
1 plays in ./experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_logic.yml
PLAY [localhost] 
2020-01-07T04:37:38.199949 (delta: 0.038383)         elapsed: 0.038383 ******** 
=============================================================================== 
2020-01-07T04:37:38.207600 (delta: 0.007595)         elapsed: 0.046034 ******** 
TASK [Gathering Facts] 
task path: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_logic.yml:2
META: ran handlers
2020-01-07T04:37:43.401942 (delta: 5.194321)         elapsed: 5.240376 ******** 
TASK [Identify the chaos util to be invoked] 
task path: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_prerequisites.yml:1
changed: [127.0.0.1] => {"changed": true, "checksum": "3936b6e8d2254d68e4fc94aeaec46d28009d50df", "dest": "./chaosutil.yml", "gid": 0, "group": "root", "md5sum": "a43553aab677c0a06c7e42881d10ad1c", "mode": "0644", "owner": "root", "size": 52, "src": "/root/.ansible/tmp/ansible-tmp-1578371863.43-275687138633183/source", "state": "file", "uid": 0}
2020-01-07T04:37:43.884838 (delta: 0.482867)         elapsed: 5.723272 ******** 
TASK [include_vars] 
task path: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_logic.yml:22
ok: [127.0.0.1] => {"ansible_facts": {"c_util": "/chaoslib/litmus/cpu_hog/pod_cpu_hog.yml"}, "ansible_included_var_files": ["/experiments/generic/pod_cpu_hog/chaosutil.yml"], "changed": false}
2020-01-07T04:37:43.937039 (delta: 0.052169)         elapsed: 5.775473 ******** 
2020-01-07T04:37:43.976317 (delta: 0.039245)         elapsed: 5.814751 ******** 
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1
2020-01-07T04:37:44.031118 (delta: 0.054761)         elapsed: 5.869552 ******** 
TASK [Generate the chaos result CR to reflect SOT (Start of Test)] 
task path: /utils/runtime/update_chaos_result_resource.yml:3
changed: [127.0.0.1] => {"changed": true, "checksum": "f4aa9698af611fdd8c80ad7e6c1f9946216a06a7", "dest": "./chaos-result.yaml", "gid": 0, "group": "root", "md5sum": "2945be817c28620c0aed9c902c7b83a8", "mode": "0644", "owner": "root", "size": 300, "src": "/root/.ansible/tmp/ansible-tmp-1578371864.07-261441195124598/source", "state": "file", "uid": 0}
2020-01-07T04:37:44.301237 (delta: 0.270088)         elapsed: 6.139671 ******** 
TASK [Apply the chaos result CR] 
task path: /utils/runtime/update_chaos_result_resource.yml:13
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f chaos-result.yaml -n default", "delta": "0:00:01.110621", "end": "2020-01-07 04:37:45.660810", "failed_when_result": false, "rc": 0, "start": "2020-01-07 04:37:44.550189", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/pod-cpu-hog unchanged", "stdout_lines": ["chaosresult.litmuschaos.io/pod-cpu-hog unchanged"]}
2020-01-07T04:37:45.701170 (delta: 1.399899)         elapsed: 7.539604 ******** 
2020-01-07T04:37:45.734334 (delta: 0.033132)         elapsed: 7.572768 ******** 
2020-01-07T04:37:45.769073 (delta: 0.034699)         elapsed: 7.607507 ******** 
included: /utils/common/status_app_pod.yml for 127.0.0.1
2020-01-07T04:37:45.819418 (delta: 0.050305)         elapsed: 7.657852 ******** 
TASK [Checking whether application pods are in running state] 
task path: /utils/common/status_app_pod.yml:2
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n default -l name=hello -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.694928", "end": "2020-01-07 04:37:46.649790", "rc": 0, "start": "2020-01-07 04:37:45.954862", "stderr": "", "stderr_lines": [], "stdout": "Running\nRunning", "stdout_lines": ["Running", "Running"]}
2020-01-07T04:37:46.690828 (delta: 0.871367)         elapsed: 8.529262 ******** 
TASK [Get the container status of application.] 
task path: /utils/common/status_app_pod.yml:9
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n default -l name=hello --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.705196", "end": "2020-01-07 04:37:47.536639", "rc": 0, "start": "2020-01-07 04:37:46.831443", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}
2020-01-07T04:37:47.577467 (delta: 0.886607)         elapsed: 9.415901 ******** 
included: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml for 127.0.0.1
2020-01-07T04:37:47.661663 (delta: 0.084155)         elapsed: 9.500097 ******** 
TASK [Select the app pod] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:6
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod -l name=hello -n default -o=custom-columns=:metadata.name --no-headers | shuf | head -1", "delta": "0:00:00.710288", "end": "2020-01-07 04:37:48.517045", "rc": 0, "start": "2020-01-07 04:37:47.806757", "stderr": "", "stderr_lines": [], "stdout": "hello-deploy-54bd75cf86-zzd2t", "stdout_lines": ["hello-deploy-54bd75cf86-zzd2t"]}
2020-01-07T04:37:48.559580 (delta: 0.897884)         elapsed: 10.398014 ******* 
TASK [Record app pod name] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:15
ok: [127.0.0.1] => {"ansible_facts": {"app_pod": "hello-deploy-54bd75cf86-zzd2t"}, "changed": false}
2020-01-07T04:37:48.616555 (delta: 0.056939)         elapsed: 10.454989 ******* 
TASK [Identify the application node] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:24
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pod hello-deploy-54bd75cf86-zzd2t -n default --no-headers -o custom-columns=:spec.nodeName", "delta": "0:00:00.721516", "end": "2020-01-07 04:37:49.472554", "rc": 0, "start": "2020-01-07 04:37:48.751038", "stderr": "", "stderr_lines": [], "stdout": "gke-playground-default-pool-37e10f0d-3vqb", "stdout_lines": ["gke-playground-default-pool-37e10f0d-3vqb"]}
2020-01-07T04:37:49.510252 (delta: 0.893663)         elapsed: 11.348686 ******* 
TASK [set_fact] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:32
ok: [127.0.0.1] => {"ansible_facts": {"app_node": "gke-playground-default-pool-37e10f0d-3vqb"}, "changed": false}
2020-01-07T04:37:49.568708 (delta: 0.058422)         elapsed: 11.407142 ******* 
TASK [Generate a run id if not passed from the engine/experiment] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:36
changed: [127.0.0.1] => {"changed": true, "cmd": "echo $(mktemp) | cut -d '.' -f 2", "delta": "0:00:00.557886", "end": "2020-01-07 04:37:50.256031", "rc": 0, "start": "2020-01-07 04:37:49.698145", "stderr": "", "stderr_lines": [], "stdout": "ksBI5l3qnu", "stdout_lines": ["ksBI5l3qnu"]}
2020-01-07T04:37:50.294774 (delta: 0.726031)         elapsed: 12.133208 ******* 
TASK [set_fact] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:40
ok: [127.0.0.1] => {"ansible_facts": {"run_id": "ksbi5l3qnu"}, "changed": false}
2020-01-07T04:37:50.353333 (delta: 0.058523)         elapsed: 12.191767 ******* 
2020-01-07T04:37:50.395909 (delta: 0.042542)         elapsed: 12.234343 ******* 
2020-01-07T04:37:50.469946 (delta: 0.073997)         elapsed: 12.30838 ******** 
TASK [Obtain Container ID of the target container] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:59
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl get pods hello-deploy-54bd75cf86-zzd2t -n default -o jsonpath='{.status.containerStatuses[?(@.name==\"flux-test\")].containerID}' | awk -F 'docker://' '{print $2}'", "delta": "0:00:00.681936", "end": "2020-01-07 04:37:51.295189", "rc": 0, "start": "2020-01-07 04:37:50.613253", "stderr": "", "stderr_lines": [], "stdout": "a888f7c9ca909f2474a2efc3b341d71d08be6918d5dc37101b0c0c0a8a9d397f", "stdout_lines": ["a888f7c9ca909f2474a2efc3b341d71d08be6918d5dc37101b0c0c0a8a9d397f"]}
2020-01-07T04:37:51.332888 (delta: 0.862897)         elapsed: 13.171322 ******* 
TASK [Record the app_container_id] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:68
ok: [127.0.0.1] => {"ansible_facts": {"c_container_id": "a888f7c9ca909f2474a2efc3b341d71d08be6918d5dc37101b0c0c0a8a9d397f"}, "changed": false}
2020-01-07T04:37:51.399299 (delta: 0.066376)         elapsed: 13.237733 ******* 
TASK [Derive chaos duration in seconds (from ms)] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:72
ok: [127.0.0.1] => {"ansible_facts": {"c_duration_sec": "120"}, "changed": false}
2020-01-07T04:37:51.463561 (delta: 0.064228)         elapsed: 13.301995 ******* 
TASK [Patch the chaos params to cpu stress job template] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:76
changed: [127.0.0.1] => {"changed": true, "checksum": "baa285317c33446042f1d9d2750bc6a30717c40a", "dest": "/chaoslib/litmus/cpu_hog/app_cpu_stress.yml", "gid": 0, "group": "root", "md5sum": "8e1a65cd0a77cc9b7cb1e4ca9dce8b3e", "mode": "0644", "owner": "root", "size": 1243, "src": "/root/.ansible/tmp/ansible-tmp-1578371871.51-93599855088887/source", "state": "file", "uid": 0}
2020-01-07T04:37:51.743057 (delta: 0.279457)         elapsed: 13.581491 ******* 
TASK [Setup cpu chaos infrastructure] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:83
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl create -f /chaoslib/litmus/cpu_hog/app_cpu_stress.yml -n default", "delta": "0:00:00.805495", "end": "2020-01-07 04:37:52.683409", "rc": 0, "start": "2020-01-07 04:37:51.877914", "stderr": "", "stderr_lines": [], "stdout": "job.batch/app-cpu-stress-ksbi5l3qnu created", "stdout_lines": ["job.batch/app-cpu-stress-ksbi5l3qnu created"]}
2020-01-07T04:37:52.734173 (delta: 0.991081)         elapsed: 14.572607 ******* 
FAILED - RETRYING: Wait until the cpu stress job is completed (240 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (239 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (238 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (237 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (236 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (235 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (234 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (233 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (232 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (231 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (230 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (229 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (228 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (227 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (226 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (225 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (224 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (223 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (222 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (221 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (220 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (219 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (218 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (217 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (216 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (215 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (214 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (213 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (212 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (211 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (210 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (209 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (208 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (207 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (206 retries left).
FAILED - RETRYING: Wait until the cpu stress job is completed (205 retries left).
TASK [Wait until the cpu stress job is completed] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:90
changed: [127.0.0.1] => {"attempts": 37, "changed": true, "cmd": "kubectl get pods -l job-name=app-cpu-stress-ksbi5l3qnu --no-headers -n default --no-headers -o custom-columns=:status.phase", "delta": "0:00:00.695419", "end": "2020-01-07 04:39:57.146970", "rc": 0, "start": "2020-01-07 04:39:56.451551", "stderr": "", "stderr_lines": [], "stdout": "Succeeded", "stdout_lines": ["Succeeded"]}
2020-01-07T04:39:57.189356 (delta: 124.455143)         elapsed: 139.02779 ***** 
TASK [temp sleep] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:104
changed: [127.0.0.1] => {"changed": true, "cmd": "sleep 300", "delta": "0:05:00.577854", "end": "2020-01-07 04:44:57.896948", "rc": 0, "start": "2020-01-07 04:39:57.319094", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
2020-01-07T04:44:57.941681 (delta: 300.752281)         elapsed: 439.780115 **** 
TASK [Tear down cpu chaos infrastructure] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:107
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl delete -f /chaoslib/litmus/cpu_hog/app_cpu_stress.yml -n default", "delta": "0:00:00.690448", "end": "2020-01-07 04:44:58.776923", "rc": 0, "start": "2020-01-07 04:44:58.086475", "stderr": "", "stderr_lines": [], "stdout": "job.batch \"app-cpu-stress-ksbi5l3qnu\" deleted", "stdout_lines": ["job.batch \"app-cpu-stress-ksbi5l3qnu\" deleted"]}
2020-01-07T04:44:58.818963 (delta: 0.877194)         elapsed: 440.657397 ****** 
TASK [Confirm that the cpu stress job is deleted successfully] 
task path: /chaoslib/litmus/cpu_hog/pod_cpu_hog.yml:113
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -l job-name=app-cpu-stress-ksbi5l3qnu --no-headers -n default", "delta": "0:00:00.704943", "end": "2020-01-07 04:44:59.663165", "rc": 0, "start": "2020-01-07 04:44:58.958222", "stderr": "No resources found.", "stderr_lines": ["No resources found."], "stdout": "", "stdout_lines": []}
2020-01-07T04:44:59.705133 (delta: 0.886131)         elapsed: 441.543567 ****** 
included: /utils/common/status_app_pod.yml for 127.0.0.1
2020-01-07T04:44:59.759875 (delta: 0.054706)         elapsed: 441.598309 ****** 
TASK [Checking whether application pods are in running state] 
task path: /utils/common/status_app_pod.yml:2
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pods -n default -l name=hello -o custom-columns=:.status.phase --no-headers", "delta": "0:00:00.711517", "end": "2020-01-07 04:45:00.610887", "rc": 0, "start": "2020-01-07 04:44:59.899370", "stderr": "", "stderr_lines": [], "stdout": "Running\nRunning", "stdout_lines": ["Running", "Running"]}
2020-01-07T04:45:00.653804 (delta: 0.893894)         elapsed: 442.492238 ****** 
TASK [Get the container status of application.] 
task path: /utils/common/status_app_pod.yml:9
changed: [127.0.0.1] => {"attempts": 1, "changed": true, "cmd": "kubectl get pod -n default -l name=hello --no-headers -o jsonpath='{.items[*].status.containerStatuses[*].ready}' | tr ' ' '\\n' | uniq", "delta": "0:00:00.719936", "end": "2020-01-07 04:45:01.518823", "rc": 0, "start": "2020-01-07 04:45:00.798887", "stderr": "", "stderr_lines": [], "stdout": "true", "stdout_lines": ["true"]}
2020-01-07T04:45:01.598251 (delta: 0.944409)         elapsed: 443.436685 ****** 
TASK [set_fact] 
task path: /experiments/generic/pod_cpu_hog/pod_cpu_hog_ansible_logic.yml:62
ok: [127.0.0.1] => {"ansible_facts": {"flag": "pass"}, "changed": false}
2020-01-07T04:45:01.654126 (delta: 0.055846)         elapsed: 443.49256 ******* 
included: /utils/runtime/update_chaos_result_resource.yml for 127.0.0.1
2020-01-07T04:45:01.714344 (delta: 0.060183)         elapsed: 443.552778 ****** 
2020-01-07T04:45:01.750735 (delta: 0.036355)         elapsed: 443.589169 ****** 
2020-01-07T04:45:01.790473 (delta: 0.039696)         elapsed: 443.628907 ****** 
TASK [Update the chaos result CR to reflect EOT (End of Test)] 
task path: /utils/runtime/update_chaos_result_resource.yml:23
changed: [127.0.0.1] => {"changed": true, "checksum": "7cbe4ab43cd9ec4df624fe23e16067ca578cd71c", "dest": "./chaos-result.yaml", "gid": 0, "group": "root", "md5sum": "c9bfb4aa5b59aeba79f395f6704e47e5", "mode": "0644", "owner": "root", "size": 297, "src": "/root/.ansible/tmp/ansible-tmp-1578372301.83-199718758864869/source", "state": "file", "uid": 0}
2020-01-07T04:45:02.662700 (delta: 0.87216)         elapsed: 444.501134 ******* 
TASK [Apply the chaos result CR] 
task path: /utils/runtime/update_chaos_result_resource.yml:33
changed: [127.0.0.1] => {"changed": true, "cmd": "kubectl apply -f chaos-result.yaml -n default", "delta": "0:00:00.856684", "end": "2020-01-07 04:45:03.666068", "failed_when_result": false, "rc": 0, "start": "2020-01-07 04:45:02.809384", "stderr": "", "stderr_lines": [], "stdout": "chaosresult.litmuschaos.io/pod-cpu-hog configured", "stdout_lines": ["chaosresult.litmuschaos.io/pod-cpu-hog configured"]}
META: ran handlers
META: ran handlers
PLAY RECAP 
127.0.0.1                  : ok=32   changed=19   unreachable=0    failed=0   

2020-01-07T04:45:03.695225 (delta: 1.032487)         elapsed: 445.533659 ****** 
=============================================================================== 
```

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml (hunk 1 above)
Comment (ksatchit) [reply chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: fixed!

File: experiments/generic/pod_cpu_hog/README.md (hunk 2 above)
Comment (ksatchit) [reply experiments/generic/pod_cpu_hog/README.md]: I guess other experiments too are using "Here". Click Here seems better - can we do it in one shot for all?

Comment (chandankumar4) [reply experiments/generic/pod_cpu_hog/README.md]: okay

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml (hunk 1 above)
Comment (ksatchit) [reply chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: cc: @shubhamchaudhary - hope this is in line with your latest pattern/changes

Comment (shubhamchaudhary) [reply chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: a_ns is the same as app_ns for this experiment.

Comment (shubhamchaudhary) [reply chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: but if we want to use this lib for openebs experiment then a_ns and app_ns may vary

File: experiments/generic/pod_cpu_hog/pod_cpu_hog_k8s_job.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+          - name: TARGET_CONTAINER
+            value: ''
+
+          - name: CPU_CORES
+            value: '1'
+
+          - name: TOTAL_CHAOS_DURATION
+            value: '60000' # in ms
+
```
Comment (uditgaurav) [inline experiments/generic/pod_cpu_hog/pod_cpu_hog_k8s_job.yml]: provide lib - [supported platforms]
Also, we can add a comment on TARGET_CONTAINER provide target container name

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+          args:
+            executable: /bin/bash
+          when: cpu_app_deploy_result.rc == 0
+        
+        - name: Confirm that the cpu stress job is not present
+          shell: >
+            kubectl get pods -l job-name=app-cpu-stress-{{ run_id }} --no-headers -n {{ app_ns }}
+          args:
+            executable: /bin/bash
+          register: result
```
Comment (uditgaurav) [inline chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: we can write a different register variable according to the task `result` is being used three times and we are just overwriting it if I'm not wrong.

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml (hunk 1 above)
Comment (ksatchit) [reply chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: in which case we can provide app_ns: "openebs" while doing the include_tasks right?

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml (hunk 4 above)
Comment (ksatchit) [reply chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: if its scope is within this block - and we are not referring it elsewhere it should be fine?

File: experiments/generic/pod_cpu_hog/pod_cpu_hog_k8s_job.yml (hunk 3 above)
Comment (ksatchit) [reply experiments/generic/pod_cpu_hog/pod_cpu_hog_k8s_job.yml]: This experiment is not platform specific (app/container level cpu chaos..). However, added a not on the LIB and also target container

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+          | shuf | head -1 
+        args:
+          executable: /bin/bash
+        register: app_pod_name
+
+      - name: Record app pod name
+        set_fact:
+          app_pod: "{{ app_pod_name.stdout }}"
+
+      when: "app_pod is not defined"
```
Comment (shubhamchaudhary) [inline chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: Can we  this check  `when: "app_pod is not defined or app_pod == '' "` ?

File: experiments/generic/pod_cpu_hog/pod_cpu_hog_k8s_job.yml [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  generateName: pod-cpu-hog- 
+spec:
+  template:
+    metadata:
+      labels:
+        experiment: pod-cpu-hog
+    spec:
+      # Placeholder that is updated by the executor for automated runs
+      # Provide appropriate SA (with desired permissions) if executed manually
+      serviceAccountName: nginx
```
Comment (shubhamchaudhary) [inline experiments/generic/pod_cpu_hog/pod_cpu_hog_k8s_job.yml]: Can we provide %CHAOS_SERVICE_ACCOUNT% here

File: chaoslib/litmus/cpu_hog/pod_cpu_hog.yml (hunk 5 above)
Comment (ksatchit) [reply chaoslib/litmus/cpu_hog/pod_cpu_hog.yml]: done!

File: experiments/generic/pod_cpu_hog/pod_cpu_hog_k8s_job.yml (hunk 6 above)
Comment (ksatchit) [reply experiments/generic/pod_cpu_hog/pod_cpu_hog_k8s_job.yml]: done!

**Pull request merged**